test_util/cluster.py

```diff
@@ -278,6 +361,105 @@ def install_dcos(
         installer.postflight()
 
 
+def upgrade_dcos(cluster, installer_url, add_config_path=None):
+
+    def master_upgrade_order(cluster):
+        """Return a list of masters with the ZooKeeper leader last."""
+        return [master_zk_mode[0] for master_zk_mode in sorted(
+            cluster.master_zk_modes(),
+            # False < True
+            key=lambda host_zk_mode: host_zk_mode[1] == 'leader',
+        )]
+
+    def upgrade_host(tunnel, role, bootstrap_url):
+        # Download the install script for the new DC/OS.
+        tunnel.remote_cmd(['curl', '-O', bootstrap_url + '/dcos_install.sh'])
+
+        # Remove the old DC/OS.
+        tunnel.remote_cmd(['sudo', '-i', '/opt/mesosphere/bin/pkgpanda', 'uninstall'])
```

Comment:
If all it does is prepend `sudo` then that seems pretty low value. And it'd need a flag for whether to pass `-i`...it doesn't seem worth the indirection.